nsot/models.py

```diff
@@ -1764,12 +1794,26 @@ def required_attributes_changed(sender, instance, action, reverse, model,
     to a ProtocolType.required_attributes.
     """
     if action == 'pre_add':
+        # import pdb; pdb.set_trace()
+        # First filter in Protocol attributes.
         attrs = model.objects.filter(pk__in=pk_set)
         if attrs.exclude(resource_name='Protocol').exists():
             raise exc.ValidationError({
                 'required_attributes': 'Only Protocol attributes are allowed'
             })
 
+        # Then make sure that they match the site of the incoming instance.
+        site_id = instance.site_id
+        wrong_site = attrs.exclude(site_id=site_id)
+        if wrong_site.exists():
+            bad_attr = wrong_site.first()
+            raise exc.ValidationError({
+                'required_attributes': (
+                    'Attributes must share the same site as '
+                    'ProtocolType.site. Got: %s' % bad_attr
```

Comment:
Instead of just reporting the first bad attribute, could you report them as a list? That would improve the UX of this, so that the user doesn't have to discover their bad attributes one-by-one

Reply:
That is a good suggestion.